question-service/src/routes/questionsRoute.ts

```diff
@@ -0,0 +1,150 @@
+import { Router, Request, Response } from 'express'
+import { Question } from '../models/questionModel'
+
+const router: Router = Router()
+
+// retrieves entire list of questions
+router.get('/', async (req: Request, res: Response) => {
+    try {
+        const questions = await Question.find({}).sort({ title: 1 })
+
+        return res.status(200).send({
+            length: questions.length,
+            data: questions
+        })
+    } catch (error) {
+        console.log(error)
+        return res.status(500).send({
+            message: "Error retrieving list of questions"
+        })
+    }
+})
+
+// retrieves a specific question by id
+router.get('/:id', async (req: Request, res: Response) => {
+    const id = req.params.id
+    
+    try {
+        const question = await Question.findById(id)
+
+        return res.status(200).send(question)
+
+    } catch (error) {
+        console.log(error)
+        return res.status(500).send({
+            message: `Error retrieving question of id: ${id}`
+        })
+    }
+})
+
+// create a question
+router.post('/', async (req: Request, res: Response) => {
+    const question = req.body
+    if (
+        !question.title ||
+        !question.difficulty ||
+        !question.description
+    ) {
+        return res.status(400).send({
+            message: "Title, difficulty, and description must not be empty"
+        })
+    }
+    
+    try {
+        await Question.create(question)
+        return res.status(200).send({
+            message: "Question successfully created"
+        }) 
+    } catch (error) {
+        console.log(error)
+        return res.status(500).send({
+            message: "Error creating question"
+        })
+    }
+})
+
+// updates a question, identifed by id
+router.put('/:id', async (req: Request, res: Response) => {
+    const id = req.params.id
+    const question = req.body
+    if (
+        !question.title ||
+        !question.difficulty ||
+        !question.description
+    ) {
+        return res.status(400).send({
+            message: "Title, difficulty, and description must not be empty"
+        })
+    }
```

Comment:
for the PUT request to update the data, why do we need to include all (title, description, difficulty) if i just want to update the difficulty